system/t07_serve/__init__.py

```diff
@@ -8,8 +8,43 @@
 import subprocess
 import shlex
 import time
+import errno
 
 from lib import BaseTest
+from socket import error as socket_error
+
+class RootDirInaccessible(BaseTest):
+    """
+    serve command aborts if rootDir is inaccessible
+    """
+    fixtureDB = False
+    fixturePool = False
+
+    configOverride = {
+        "rootDir": "/root" # any directory that exists but is not writable
+    }
+    runCmd = "aptly serve -listen=127.0.0.1:8765"
+
+    def run(self):
+        try:
+            proc = subprocess.Popen(shlex.split(self.runCmd), stderr=subprocess.STDOUT, stdout=subprocess.PIPE, bufsize=0)
```

Comment:
I was looking for a way to `expectedCode = 1` without luck - didn't know it's so easy.